src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -384,14 +397,27 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     // Parallel compilation is unstable so is disabled by default.
     const bool par_opt = m_cfg.get<::intel_npu::NPUW_PARALLEL_COMPILE>();
     if (par_opt) {
-        ov::parallel_for(orderedSubgraphs.size(), compile);
+        ov::parallel_for(idx_subgraph_to_compile.size(), compile);
     } else {
         // TODO: Introduce npuw::serial(i, f) instead where f is a _funcall
-        for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        for (std::size_t i = 0u; i < idx_subgraph_to_compile.size(); i++) {
             compile(i);
         }
     }
 
+    // Process skipped submodels
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        if (orderedSubgraphs[i]._optimized_out) {
+            continue;  // do nothing here
+        }
+
+        const std::size_t real_id = m_compiled_submodels[i].replaced_by.value_or(i);
+        if (m_compiled_submodels[i].replaced_by && real_id != i) {
+            m_compiled_submodels[i].compiled_model = m_compiled_submodels[real_id].compiled_model;
+            m_compiled_submodels[i].device_it = m_compiled_submodels[real_id].device_it;
```

Comment:
we didn't do this before. Why do we do this now?

Reply:
Removed, thanks

Reply:
Does it work now? :)

Reply:
It does!